### PR TITLE
Implement Beam SelectPrivatePartitions

### PR DESCRIPTION
**Pull request by Kulankhina**

### CODE REVIEW

File: pipeline_dp/private_beam.py [hunk 1]
```diff
@@ -187,6 +187,31 @@ def expand(self, pcol: pvalue.PCollection) -> pvalue.PCollection:
         return dp_result
 
 
+class SelectPrivatePartitions(PrivatePTransform):
+    """Transform class for computing a collection of PrivatePCollection partition keys using DP."""
+
+    def __init__(self,
+        select_partitions_params: aggregate_params.SelectPrivatePartitionsParams,
+        partition_extractor: Callable,
+        label: Optional[str] = None):
```
Comment (dvadym) [inline pipeline_dp/private_beam.py:196]: label is not optional for Beam, please remove =None

File: pipeline_dp/dp_engine.py [hunk 2]
```diff
@@ -156,7 +156,9 @@ def sample_unique_elements_fn(pid_and_pks):
 
             return ((pid, pk) for pk in sampled_elements)
 
-        col = self._ops.flat_map(col, sample_unique_elements_fn)
+        col = self._ops.flat_map(col,
```
Comment (dvadym) [inline pipeline_dp/dp_engine.py:159]: thanks!

**Review by dvadym — COMMENTED** [review]
Thanks, it looks great! Just a minor comment

File: tests/private_beam_test.py [hunk 3]
```diff
@@ -239,6 +239,71 @@ def test_privacy_id_count_calls_aggregate_with_params(self, mock_aggregate):
...
+            private_collection | transformer
+
+            # Assert
+            self.assertEqual(transformer._budget_accountant, budget_accountant)
+            mock_select_private_partitions.assert_called_once()
+
+            args = mock_select_private_partitions.call_args[0]
+            self.assertEqual(args[1], select_partitions_params)
+
+    def test_select_private_partitions_returns_sensible_result(self):
```
Comment (Kulankhina) [inline tests/private_beam_test.py:276]: This test is failing with TypeError

File: pipeline_dp/private_beam.py [hunk 4]
```diff
@@ -187,6 +187,30 @@ def expand(self, pcol: pvalue.PCollection) -> pvalue.PCollection:
         return dp_result
 
 
+class SelectPrivatePartitions(PrivatePTransform):
```
Comment (Kulankhina) [inline pipeline_dp/private_beam.py:190]: Don't push this. I forgot budget_weight.

Comment (Kulankhina) [reply pipeline_dp/private_beam.py]: I'll do it in the next PR

**Review by dvadym — APPROVED** [review]
Thanks a lot for implementing select private partitions!

**Pull request merged**